impress-org/givewp-next-gen
# Refactor: v3 gateway api compatibility

## Description

Depends on: https://github.com/impress-org/givewp/pull/6830



This updates gateways for v3 forms compatibility. 

- Removed the `NextGenPaymentGatewayInterface` and any check for it.  v3 forms now use the $supportedFormVersion filter `give(PaymentGatewayRegister::class)->getPaymentGateways(3)` to retrieve gateways that support v3 forms.
- Simplified `enqueueScript()` to use WordPress natively
- Renamed NextGen specific test gateways to be generic (migrating these to core)
- Both Test Gateway and Test Gateway Offsite classes have been merged into core but registering them & loading their js file here.

## Affects



- All our v3 compatible gateways

## Visuals



N/A

## Testing Instructions



Make sure you're on GiveWP https://github.com/impress-org/givewp/pull/6830 or GiveWP develop if merged

Just make sure v3 compatible gateways show up on the form properly.

## Pre-review Checklist
